ansible/roles/lvm/tasks/manage_lvm.yml

```
@@ -67,7 +68,9 @@
   when: lvm__logical_volumes|d(False) and
         item.vg|d() and item.lv|d() and item.size|d() and
         item.state|d('present') != 'absent' and
-        ((item.mount|d() and item.fs|d(True)) or item.fs|d())
+        ((item.mount|d() and item.fs|d(True)) or item.fs|d()) and
+        (not ansible_check_mode or
+         (ansible_check_mode and not lvm__register_logical_volumes.changed))
```

I propose this syntax for two reason:

* https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html
* Pythonic/PEP

```suggestion
         (ansible_check_mode and lvm__register_logical_volumes is not changed))
```

---

Thanks for the suggestion, I like your style :)